tracing/test-log-support/tests/log_enabled_no_trace.rs

```diff
@@ -0,0 +1,46 @@
+use test_log_support::Test;
+use tracing::{enabled, error, warn, Level};
+
+#[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
+#[test]
+fn test_log_enabled() {
```

Comment:
We could add tests like that. Unfortunately, I think the easiest way to do this would be to add an additional test crate; the reason the `test-log-support` crate exists is to isolate the `log-always` feature flag from the rest of `tracing`, so that the tests are always run with the feature enabled. If we wanted to do the same thing for `log` but not `log-enabled`, we would need to add an additional crate.

If you're willing to put the effort into doing that, more tests is always better...

Reply:
would it be reasonable to instead update the existing crate so that `log-always` is only turned on when some feature is enabled, and then update CI to run the tests both with and without that feature? this would be nice so we could share the helper code to assert on log messages

Reply:
we can share the helper code by factoring it out into an additional crate that's a path dep of all test crates. i would prefer not to add a feature flag to `tracing` that's only used for internal testing purposes, since that would be exposed publicly to all users of `tracing`...

Reply:
...oh, i misinterpreted what you meant, sorry --- i think you actually meant adding a feature flag to the _test crate_. that would be a fine approach, too. 

i think if we do that, we probably ought to add a README or something to the `test-log-support` crate that summarizes how to run the tests with different feature combinations, so that developers can know how to run those tests locally as well as on CI. if we document it, i'm fine with adding a feature flag to the test crate.